src/aws_encryption_sdk/materials_managers/default.py

```diff
@@ -143,3 +207,55 @@ def decrypt_materials(self, request):
         )
 
         return DecryptionMaterials(data_key=data_key, verification_key=verification_key)
+
+    def _decrypt_materials_using_keyring(self, request):
+        """Obtains a plaintext data key from one or more encrypted data keys
+        using underlying keyring.
+
+        :param request: decrypt materials request
+        :type request: aws_encryption_sdk.materials_managers.DecryptionMaterialsRequest
+        :returns: decryption materials
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :raises InvalidCryptographicMaterialsError: if keyring cannot complete decryption materials
+        :raises InvalidCryptographicMaterialsError:
+           if decryption materials received from keyring do not match request
+        """
+        verification_key = self._load_verification_key_from_encryption_context(
+            algorithm=request.algorithm, encryption_context=request.encryption_context
+        )
+        decryption_materials = DecryptionMaterials(
+            algorithm=request.algorithm,
+            encryption_context=request.encryption_context.copy(),
+            verification_key=verification_key,
+        )
+
+        final_materials = self.keyring.on_decrypt(
+            decryption_materials=decryption_materials, encrypted_data_keys=request.encrypted_data_keys
+        )
+
+        if not final_materials.is_complete:
+            raise InvalidCryptographicMaterialsError("Decryption materials are incomplete!")
+
+        materials_are_valid = (
```

Comment:
Need to have a plaintext data key...

Reply:
`is_complete`

https://github.com/aws/aws-encryption-sdk-python/blob/952bce44db630754de6a0dad451f72665f83b1c7/src/aws_encryption_sdk/materials_managers/__init__.py#L404-L420